lib/linez.ts

```diff
@@ -13,29 +12,30 @@ var boms: { [key: string]: Buffer } = {
 	'utf-32be': new Buffer([0x00, 0x00, 0xfe, 0xff])
 };
 
-function linez(contents: string): linez.Document;
-function linez(buffer: Buffer): linez.Document;
-function linez(file: string|Buffer): linez.Document {
+var globalOptions: linez.Options;
+
+function linez(contents: string, options?: linez.Options): linez.Document;
+function linez(buffer: Buffer, options?: linez.Options): linez.Document;
+function linez(file: string|Buffer, options?: linez.Options): linez.Document {
+	options = objectAssign({}, globalOptions, options);
 	if (typeof file === 'string') {
-		return new linez.Document(parseLines(file));
+		return new linez.Document(parseLines(file, options));
 	}
 	var buffer = <Buffer>file;
 	var doc = new linez.Document();
 	doc.charset = detectCharset(buffer);
 	var bom = boms[doc.charset];
 	var encoding = doc.charset.replace(/bom$/, '');
 	if (iconv.encodingExists(encoding)) {
-		doc.lines = parseLines(iconv.decode(buffer.slice(bom.length), encoding));
+		doc.lines = parseLines(iconv.decode(buffer.slice(bom.length), encoding), options);
 	} else {
-		doc.lines = parseLines(buffer.toString('utf8'));
+		doc.lines = parseLines(buffer.toString('utf8'), options);
 	}
 	return doc;
 }
 
 function detectCharset(buffer: Buffer) {
-	var bomKeys = Object.keys(boms);
-	for (var i = 0; i < bomKeys.length; i++) {
-		var charset = bomKeys[i];
+	for (var charset in boms) {
```

Comment:
Oh, right, it's an object. [`_.forEach`](https://lodash.com/docs#forEach) does it:

```js
_.forEach({ 'a': 1, 'b': 2 }, function(value, key) {
  console.log(key);
});
```